pkg/controller/dnsendpoint/nameserver/azure.go

```diff
@@ -119,21 +121,26 @@ func (q *azureQuery) recordSet(domain string, values sets.String) dns.RecordSet
 
 // queryNameServer queries Azure for the name servers for the specified domain in the specified managed zone.
 func (q *azureQuery) queryNameServer(azureClient azureclient.Client, rootDomain string, domain string) (sets.String, error) {
-	recordSets, err := azureClient.ListRecordSetsByZone(context.TODO(), q.resourceGroupName, rootDomain)
+	recordSetsPage, err := azureClient.ListRecordSetsByZone(context.TODO(), q.resourceGroupName, rootDomain, to.Int32Ptr(50))
 	if err != nil {
 		return nil, err
 	}
 
-	for _, recordSet := range *recordSets {
-		if *recordSet.Type != string(dns.NS) {
-			continue
-		}
-		if *recordSet.Name == domain {
-			values := sets.NewString()
-			for _, v := range *recordSet.NsRecords {
-				values.Insert(*v.Nsdname)
+	for recordSetsPage.NotDone() {
+		for _, recordSet := range recordSetsPage.Values() {
+			if *recordSet.Type != string(dns.NS) {
+				continue
+			}
+			if *recordSet.Name == domain {
+				values := sets.NewString()
+				for _, v := range *recordSet.NsRecords {
+					values.Insert(*v.Nsdname)
+				}
+				return values, nil
 			}
-			return values, nil
+		}
+		if err := recordSetsPage.Next(); err != nil {
```

Comment:
`Next` is deprecated. Use `NextWithContext`.

Reply:
Fixed